tests/js/JSTest.php

```diff
@@ -152,6 +152,29 @@ public function dataProvider()
             '',
         );
 
+        // preserve important comments
+        $tests[] = array(
+            '/*! This is a JS comment */',
+            '/*! This is a JS comment */',
+        );
+
+        $tests[] = array(
+            '/*!
+              * This is a multi line JS comment
+              */',
+            '/*!*This is a multi line JS comment*/',
```

Comment:
This shouldn't get "minified". Not only does the comment become less clear than it was, but there's a risk in "ignoring" the content in this comment: if there's string-like or regex-like content started inside this (ignored) comment, it could confuse the extractStrings & extractRegex methods.

A better solution would be to still remove *all* comments, but also `$this->registerPattern` a more specific pattern for these comment types, that just temporarily extracts these comments and then later on restores them (similar to extractStrings & extractRegex)

I'm happy to land this functionality, but not until this is fixed (I can probably make the required change myself soon)